swap/src/rpc/methods.rs

```diff
@@ -0,0 +1,241 @@
+use crate::api::request::{Method, Request};
+use crate::api::Context;
+use crate::bitcoin::bitcoin_address;
+use crate::monero::monero_address;
+use crate::{bitcoin, monero};
+use anyhow::Result;
+use jsonrpsee::server::RpcModule;
+use jsonrpsee::types::Params;
+use libp2p::core::Multiaddr;
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::sync::Arc;
+use uuid::Uuid;
+
+pub fn register_modules(context: Arc<Context>) -> RpcModule<Arc<Context>> {
+    let mut module = RpcModule::new(context);
+
+    module
+        .register_async_method("suspend_current_swap", |params, context| async move {
+            execute_request(params, Method::SuspendCurrentSwap, &context).await
+        })
+        .unwrap();
```

Comment:
this method should return a Result so that these unwraps can just be `?`

Reply:
The methods are inside `register_modules`, which returns a `RpcModule<Arc<Context>>`. Not sure if it would make sense to wrap it inside a Result here.

Reply:
you can just use a ? where that method is called.